src/commands/lint.cr

```diff
@@ -0,0 +1,75 @@
+module Mint
+  class Cli < Admiral::Command
+    class Lint < Admiral::Command
+      include Command
+
+      define_help description: "Lints the project for syntax and type errors."
+
+      define_flag json : Bool,
+        description: "Output errors to a JSON file",
+        default: false,
+        required: false
+
+      def run
+        execute "Linting" do
+          lint
+        end
+      end
+
+      def lint
+        sources = [] of String
+        errors = [] of Exception
+
+        ast =
+          Ast.new
+            .merge(Core.ast)
+
+        begin
+          sources =
+            Dir.glob(SourceFiles.all)
+        rescue ex
+          ex_handler errors, ex
+        end
+
+        sources.reduce(ast) do |memo, file|
+          parsed = Parser.parse(file)
+
+          if memo
+            memo.merge parsed
+          end
+        rescue ex
+          ex_handler errors, ex
+        end
+
+        if errors.empty?
+          type_checker =
+            TypeChecker.new(ast)
+
+          done = false
+
+          while !done
+            begin
+              type_checker.check
+            rescue ex
+              ex_handler errors, ex
+            else
+              done = true
+            end
+          end
+        end
+
+        if flags.json
+          puts errors.compact_map(&.message.presence).to_json
+        end
```

Comment:
This should have an else branch where it prints all the errors, otherwise STDOUT will have all the error because of `ex_handler` and then the JSON.

Reply:
Like this?
```crystal
if flags.json
  puts errors.compact_map(&.message.presence).to_json
else
  puts errors
end
```

Reply:
Almost, `errors.each { |error| puts error }` otherwise it will print the brackets and commas since it's an array.

Reply:
Gotcha, okay. I've added this 👍